x-pack/plugins/monitoring/server/routes/api/v1/_health/errors_helpers/errors_query.ts

```diff
@@ -82,11 +87,34 @@ const errorsByMetricset = {
   },
 };
 
-const errorsAggregation = {
+const errorsByDataStream = {
   terms: {
-    field: 'event.module',
+    field: 'data_stream.dataset',
   },
   aggs: {
-    errors_by_dataset: errorsByMetricset,
+    latest_docs: {
+      top_hits: {
+        sort: [
+          {
+            '@timestamp': {
+              order: 'desc',
+            },
+          },
+        ],
+        size: MAX_BUCKET_SIZE,
+        _source: {
+          includes: ['@timestamp', 'error', 'data_stream'],
+        },
+      },
+    },
   },
 };
+
+const errorsAggregation = (errorQueryIsDataStream?: boolean) => ({
+  terms: {
+    field: 'event.module',
```

Comment:
Should this field also be conditional depending on `errorQueryIsDataStream` ?

Reply:
Thank you for spotting that! I changed it.